spec/models/user/academics/registration_spec.rb

```diff
@@ -0,0 +1,157 @@
+    {
+      'term' => term,
+      'academicCareer' => academic_career,
+      'termUnits' => term_units,
+      'academicLevels' => academic_levels,
+    }
+  end
+  subject { described_class.new(data) }
+
+  describe '#term_id' do
+    it 'returns term id' do
+      expect(subject.term_id).to eq '2198'
+    end
+  end
+
+  describe '#undergraduate?' do
+    context 'when academic career is undergraduate' do
+      let(:academic_career) { {'code' => 'UGRD', 'description' => 'Undergraduate'} }
+      it 'returns true' do
+        expect(subject.undergraduate?).to eq true
+      end
+    end
+    context 'when academic career is not undergraduate' do
+      let(:academic_career) { {'code' => 'GRAD', 'description' => 'Graduate'} }
+      it 'returns false' do
+        expect(subject.undergraduate?).to eq false
+      end
+    end
+  end
+
+  describe '#career_code' do
+    it 'returns academic career code' do
+      expect(subject.career_code).to eq 'UGRD'
+    end
+  end
+
+  describe '#career_description' do
+    it 'returns academic career description' do
+      expect(subject.career_description).to eq 'Undergraduate'
+    end
+  end
+
+  describe '#unit_totals' do
+    it 'returns the total term units' do
+      result = subject.unit_totals
+      expect(result['type']['code']).to eq 'Total'
+      expect(result['unitsEnrolled']).to eq 12
+    end
+    context 'when term units not present' do
+      let(:term_units) { [] }
+      it 'returns empty hash' do
+        expect(subject.unit_totals).to eq({})
+      end
+    end
+  end
+
+  describe '#total_units_taken' do
+    it 'returns total term units taken' do
+      expect(subject.total_units_taken).to eq 4
+    end
+  end
+
+  describe '#total_units_enrolled' do
+    it 'returns total term units enrolled' do
+      expect(subject.total_units_enrolled).to eq 12
+    end
+  end
+
+  describe '#enrolled?' do
+    context 'when no units taken or enrolled' do
+      let(:term_units_total_enrolled) { 0 }
+      let(:term_units_total_taken) { 0 }
+      it 'returns false' do
+        expect(subject.enrolled?).to eq false
+      end
+    end
+    context 'when units taken but not enrolled' do
+      let(:term_units_total_enrolled) { 0 }
+      let(:term_units_total_taken) { 4 }
+      it 'returns true' do
+        expect(subject.enrolled?).to eq true
+      end
+    end
+    context 'when units enrolled but not taken' do
+      let(:term_units_total_enrolled) { 12 }
+      let(:term_units_total_taken) { 0 }
+      it 'returns true' do
+        expect(subject.enrolled?).to eq true
+      end
+    end
+  end
+
+  describe '#academic_levels' do
+    it 'returns academic level objects' do
+      result = subject.academic_levels
+      expect(result.count).to eq 2
+      expect(result[0]).to be_an_instance_of ::User::Academics::Level
+      expect(result[1]).to be_an_instance_of ::User::Academics::Level
+      expect(result[0].type_code).to eq 'BOT'
+      expect(result[1].type_code).to eq 'EOT'
```

Comment:
I'm not sure you really need to be concerned about the type of objects. You could express your intent more concisely with `expect(result.collect(&:type_code)).to eq ['BOT', 'EOT']`